src/gameComponents/HullZone.java

```diff
@@ -0,0 +1,148 @@
+package gameComponents;
+
+import org.joml.Vector2f;
+
+public class HullZone {
+
+    /*
+    A hull zone is a collection of space on a ship that stores shields attack values and some location data.
+    A hull zone can take shield damage without outside assistance
+     */
+
+    //Available enums for a ship facing. AUX arcs exist for huge ships, otherwise use the normal ones.
+    public enum Facing {FRONT, LEFT, RIGHT, BACK, AUX_LEFT, AUX_RIGHT}
+
+    //The named facing of the ship
+    private Facing facing;
+
+    //The anti-ship battery values
+    private int redBattery;
+    private int blueBattery;
+    private int blackBattery;
+
+    //The current and maximum shield value of the hull zone. In most cases these start identical
+    private int shields;
+    private int shieldsMaximum;
+
+    //The Line of sight offset relative to ship center for a HullZone
+    private Vector2f LoSOffset;
+
+    //The left/right origin offsets relative to ship center for a HullZone
+    private Vector2f leftArcOffset;
+    private Vector2f rightArcOffset;
+
+    //The left right arc angles for a ship.
+    private float leftArcAngle;
+    private float rightArcAngle;
+
+    //No null constructor
+
+    //Constructor from string looks too complex to make an attempt with how the ship parser is laid out
+
+    /**
+     * Valued constructor
+     * @param facing facing
+     * @param redBattery red dice
+     * @param blueBattery blue dice
+     * @param blackBattery black dice
+     * @param shields shields
+     * @param LoSOffset .
+     * @param leftArcOffset .
+     * @param leftArcAngle .
+     * @param rightArcOffset .
+     * @param rightArcAngle .
+     */
+
+    public HullZone(Facing facing, int redBattery, int blueBattery,int blackBattery, int shields, Vector2f LoSOffset, Vector2f leftArcOffset, float leftArcAngle, Vector2f rightArcOffset, float rightArcAngle){
+        this.facing = facing;
+        this.redBattery = redBattery;
+        this.blueBattery = blueBattery;
+        this.blackBattery = blackBattery;
+        this.shields = shields;
+        this.shieldsMaximum = shields;
+        this.LoSOffset = LoSOffset;
+        this.leftArcOffset = leftArcOffset;
+        this.leftArcAngle = leftArcAngle;
+        this.rightArcAngle = rightArcAngle;
+        this.rightArcOffset = rightArcOffset;
+    }
```

Comment:
Not wild about having so many things in the constructor.

I'd anticipated a list like the parsed string for armament, shields, a polygon, and a parent ship.  A Facing name makes sense too.

Reply:
This will be a later refactor

Reply:
an issue with a string based contructor is that we are building that string at the same time as 3-6 other hull zones, and so the parse to parse from ship parse to hull zone parse sounds messy on the initial sell

Reply:
A `Ship` object is receiving a String for armament because text file.  Its going to have to be turned into something useful at some point.  I'd prefer to have it in the `HullZone` object because armament belongs to hull zones in the physical game, and to remove some logic out of the `Ship` class which I expect to end up being pretty involved as it is.